### DISCUSSION TITLE
What the best way to show argo CD deployment status in Gitlab CI?

### DISCUSSION THREAD

**Discussion by stefankuksenko** [argoproj/argo-cd#7475] [Q&A]
At the moment I am trying to make full CI/CD using Argo Events, Argo Workflow and ArgoCD but there are a few important requirements to be done before we can implement it:
- We need to see a status of argo workflow jobs or argocd deployments in Gitlab CI to let developer know if there something went wrong on any of CI/CD steps
- We need to allow merges in Gitlab only is everything is ok on workflow and argocd side

At the moment the only option which i see is to make additiional Gitlab CI job which checks a status of workflows and argocd applications.
 What else we can do ? i was sure it is a common problem, but haven't found any discussions regarding this topic.

**Comment by reggie-k** [comment]
How about the last GitlabCI step performing ArgoCD app wait (which is synchronous, thus will fail the pipeline if the app is not synced ok), and marking on Gitlab side the option of "merge only on a successful pipeline run"?
